controller/accounts.go

```diff
@@ -41,103 +41,43 @@ func NewAccounts(
+func (a *Accounts) GetByAddress(address model.Address, projectID uuid.UUID) (*model.Account, error) {
+	account, err := a.blockchain.GetAccount(projectID, address)
 	if err != nil {
 		return nil, err
 	}
-
-	return account.
-		MergeFromStore(&acc).
-		Export(), nil
+	return account.Export(), nil
 }
 
 func (a *Accounts) AllForProjectID(projectID uuid.UUID) ([]*model.Account, error) {
-	var accounts []*model.Account
-
-	err := a.store.GetAccountsForProject(projectID, &accounts)
+	var proj model.Project
+	err := a.store.GetProject(projectID, &proj)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get accounts")
+		return nil, err
 	}
 
-	addresses := make([]model.Address, len(accounts))
-	for i, account := range accounts {
-		addresses[i] = account.Address
+	addresses := make([]model.Address, proj.NumberOfAccounts)
+	for i := 0; i < proj.NumberOfAccounts; i++ {
+		addresses[i] = getAddressForAccountIndex(i)
 	}
 
 	accs, err := a.blockchain.GetAccounts(projectID, addresses)
 	if err != nil {
 		return nil, err
 	}
 
-	exported := make([]*model.Account, len(accounts))
-	for i, account := range accounts {
-		accs[i].MergeFromStore(account)
+	exported := make([]*model.Account, proj.NumberOfAccounts)
+	for i := 0; i < proj.NumberOfAccounts; i++ {
 		exported[i] = accs[i].Export()
 	}
 
 	return exported, nil
 }
 
-func (a *Accounts) Update(input model.UpdateAccount) (*model.Account, error) {
-	if input.UpdateCode() {
-		return a.updateCode(input)
-	}
-
-	return a.deployCode(input)
-}
-
-// updateCode only updates the database code of an account.
-func (a *Accounts) updateCode(input model.UpdateAccount) (*model.Account, error) {
-	var acc model.Account
-	err := a.store.UpdateAccount(input, &acc)
-	if err != nil {
-		return nil, err
-	}
-
-	return acc.Export(), nil
-}
-
-// deployCode deploys code on the flow network.
-func (a *Accounts) deployCode(input model.UpdateAccount) (*model.Account, error) {
-	var dbAccount model.Account
-	err := a.store.GetAccount(input.ID, input.ProjectID, &dbAccount)
-	if err != nil {
-		return nil, err
-	}
-
-	flowAccount, err := a.blockchain.GetAccount(input.ProjectID, dbAccount.Address)
-	if err != nil {
-		return nil, err
-	}
-
-	// reset the state first if already contains deployed code
-	if flowAccount.HasDeployedCode() {
-		var proj model.Project
-		err := a.store.GetProject(input.ProjectID, &proj)
-		if err != nil {
-			return nil, err
-		}
-
-		_, err = a.blockchain.Reset(&proj)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	flowAccount, err = a.blockchain.DeployContract(input.ProjectID, dbAccount.Address, *input.DeployedCode)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to deploy account code")
-	}
-
-	return flowAccount.
-		MergeFromStore(&dbAccount).
-		Export(), nil
+// getAddressForAccountIndex calculates the address based on the offset from the initial account address
+func getAddressForAccountIndex(index int) model.Address {
```

Comment:
This would be better to be part of the model.Address function like `NewAddressFromIndex`

Reply:
Done